core/src/Revolution/Processors/Resource/Update.php

```diff
@@ -479,18 +477,48 @@ public function checkPublishingPermissions()
      *
      * @return boolean
      */
-    public function checkForUnPublishOnSiteStart()
+    public function checkForUnPublishOnSiteSettingPages()
     {
-        $passed = true;
+        $result = true;
+        $isSiteStart = false;
+        $isSiteErrorPage = false;
+        $isSiteUnavailablePage = false;
+        $deleted = $this->getProperty('deleted', null);
         $published = $this->getProperty('published', null);
         $publishDate = $this->getProperty('pub_date');
         $unPublishDate = $this->getProperty('unpub_date');
-        if ($this->isSiteStart && ($published !== null && empty($published))) {
-            $passed = $this->modx->lexicon('resource_err_unpublish_sitestart');
-        } else if ($this->isSiteStart && (!empty($publishDate) || !empty($unPublishDate))) {
-            $passed = $this->modx->lexicon('resource_err_unpublish_sitestart_dates');
-        }
-        return $passed;
+        $workingContext = $this->modx->getContext($this->getProperty('context_key'));
+        if ($this->getProperty('id') == $workingContext->getOption('site_start') || $this->getProperty('id') == $this->modx->getOption('site_start')) {
+            $isSiteStart = true;
+        }
+        if ($this->getProperty('id') == $workingContext->getOption('error_page') || $this->getProperty('id') == $this->modx->getOption('error_page')) {
+            $isSiteErrorPage = true;
+        }
+        if ($this->getProperty('id') == $workingContext->getOption('site_unavailable_page') || $this->getProperty('id') == $this->modx->getOption('site_unavailable_page')) {
+            $isSiteUnavailablePage = true;
+        }
+
+        if ($isSiteStart && $deleted == 1) {
+            $result = $this->modx->lexicon('resource_err_delete_sitestart');
+        } else if ($isSiteStart && $published == 0) {
```

Comment:
Replace all `else if` with a regular `if` and follow return early pattern.

`return $this->modx...`